cmd/tealdbg/debugger.go

```diff
@@ -477,9 +535,12 @@ func (d *Debugger) Update(state *logic.DebugState) error {
 
 	// copy state to prevent a data race in this the go-routine and upcoming updates to the state
 	go func(localState logic.DebugState) {
+		// Copy callstack information
+		s.setCallStack(state.CallStack)
 		// Check if we are triggered and acknowledge asynchronously
-		if cfg.BreakAtLine != noBreak {
-			if cfg.BreakAtLine == stepBreak || breakpointLine(localState.Line) == cfg.BreakAtLine {
+		if !cfg.NoBreak {
+			if cfg.StepBreak || (localState.Line == cfg.BreakAtLine &&
+				(!cfg.StepOutOver || cfg.CallDepth == len(state.CallStack))) {
```

Comment:
it is time to refactor `debugConfig ` (cfg) and make an object with methods like `isBreak(state)`, `setStepBreak()`, `setBreakAt(line)`

Reply:
Agreed, currently also refactoring the `BreakAtLine` into a set so `Resume` can also break during backward jumps.